tools/webdriver/webdriver/client.py

```diff
@@ -372,6 +372,42 @@ def from_json(cls, json, session):
         return cls(uuid, session)
 
 
+class ShadowRoot(object):
+    identifier = "shadow-075b-4da1-b6ba-e579c2d3230a"
+
+    def __init__(self, session, id):
+        """
+        Construct a new shadow root representation.
+
+        :param id: Shadow root UUID which must be unique across
+            all browsing contexts.
+        :param session: Current ``webdriver.Session``.
+        """
+        self.id = id
+        self.session = session
+
+    @classmethod
+    def from_json(cls, json, session):
+        uuid = json[ShadowRoot.identifier]
+        return cls(uuid, session)
+
+    def send_shadow_command(self, method, uri, body=None):
+        url = "shadow/%s/%s" % (self.id, uri)
```

Comment:
Please change this to 
```suggestion
        url = "shadow/{}/{}".format(self.id, uri)
```

Reply:
Done in 5720604

Reply:
or rather, botched in 5720604 and fixed in 5d3b4c5